src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java

```diff
@@ -174,22 +180,45 @@ public Response<UserResponseDTO> getStudentById(String id) {
     }
 
     /**
-     * Retrieves a list of all users from the database.
+     * Retrieves a list of all users from the database by page.
      *
      * @return A response containing a list of user information.
      */
-    public Response<List<UserResponseDTO>> getAllUsers() {
-        Response<List<UserResponseDTO>> response = new Response<>();
-        List<UserDAO> userDAOS = userRepository.findAll();
-        List<UserResponseDTO> usersList = new ArrayList<>();
-        userDAOS.forEach(userDAO -> usersList.add(this.modelMapper.map(userDAO, UserResponseDTO.class)));
-        if (usersList.isEmpty())
-            response.setMeta(Meta.builder().message("DataBase is Empty").statusCode(HttpStatus.NOT_FOUND.value()).build());
-        else
-            response.setMeta(Meta.builder().message("Retrieval Successful").statusCode(HttpStatus.OK.value()).build());
-        log.info("Total Users: {}",usersList.size());
-        log.info("Users Details: {}", usersList);
-        response.setData(usersList);
-        return response;
+    public Response<List<UserResponseDTO>> getAllUsers(int page, int size, String sortBy, String sortOrder) {
+        try {
+            Sort sort = Sort.by(Sort.Direction.fromString(sortOrder), sortBy);
+            Pageable pageable = PageRequest.of(page, size, sort);
+            Page<UserDAO> userPage = userRepository.findAll(pageable);
+
+            if (!userPage.hasContent()) {
+                Response<List<UserResponseDTO>> emptyResponse = new Response<>();
+                Meta emptyMeta = Meta.builder().message("Database is empty").statusCode(HttpStatus.NOT_FOUND.value()).build();
+                emptyResponse.setMeta(emptyMeta);
+                return emptyResponse;
+            } else {
+                List<UserResponseDTO> usersList = userPage.getContent().stream()
+                        .map(userDAO -> modelMapper.map(userDAO, UserResponseDTO.class))
+                        .collect(Collectors.toList());
+
+                Response<List<UserResponseDTO>> response = new Response<>();
+                Meta meta = Meta.builder().message("Retrieval Successful").statusCode(HttpStatus.OK.value()).build();
+                log.info("Total Users: {}",usersList.size());
+                log.info("Users Details: {}", usersList);
+                response.setMeta(meta);
+                response.setData(usersList);
+
+                return response;
+            }
+        } catch (DataAccessException ex) {
+            Response<List<UserResponseDTO>> errorResponse = new Response<>();
+            Meta meta = Meta.builder().message("Database not found").statusCode(HttpStatus.NOT_FOUND.value()).build();
```

Comment:
do we need DB not found exception separately?

Reply:
ill remove it bhaiya faizan bhaiya said its not necessary

Reply:
please make sure you understand stuff correctly its not DB not found exception you are catching here, its Data Access not found, we definitely need to catch the only issue i see here is that message is incorrect, Also @Umaid-gh please guide him on how the Exception handler works so that he does not set the message in meta rather throw IquaisException

Reply:
Also, the correct error message would be something like "No data found for the search string"